### DISCUSSION TITLE
Possible/How to set content type for UploadFile?

### DISCUSSION THREAD

**Discussion by jeremyschiemann** [fastapi/fastapi#9107] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI, File, UploadFile

app = FastAPI()


@app.get("/")
def file_upload(file: UploadFile = File(..., media_type='text/xml')):
    return {"Hello": "World"}
```

### Description

The schema created for this file doesn't mention the specified media type. (see: [OpenAPI](https://swagger.io/docs/specification/describing-request-body/multipart-requests/) 'Specifying Content-Type')
How is a application supposed to know what kind of files it can send?

I'd expect something like
```
encoding: {
        "file": {"contentType": "text/xml"}
}
```

### Environment

* OS: Windwos 10
* FastAPI Version: 0.60.1
* Python version: 3.8

**Comment by rkbeatss** [comment]
See similar #1018. Looks interesting, down to work on a fix for this!

**Comment by YuriiMotov** [comment, marked as answer]
You can add it using `openapi_extra` parameter of path decorator:

```python
from fastapi import FastAPI, File, UploadFile

app = FastAPI()

@app.post(
    "/",
    openapi_extra={
        "requestBody": {
            "content": {
                "multipart/form-data": {
                    "encoding": {"file": {"contentType": "text/xml"}}
                }
            }
        }
    },
)
def file_upload(file: UploadFile = File(...)):
    return {"Hello": "World"}
```

The schema of request body looks like:

```json
       "requestBody": {
          "content": {
            "multipart/form-data": {
              "schema": {
                "$ref": "#/components/schemas/Body_file_upload__post"
              },
              "encoding": {
                "file": {
                  "contentType": "text/xml"
                }
              }
            }
          },
          "required": true
        },
```

Unfortunately it doesn't take any effect in neither Swagger UI nor Redoc
